FactoryWeather/App/VIP Settings/SettingsViewController.swift

```diff
@@ -0,0 +1,197 @@
+//
+//  SettingsViewController.swift
+//  FactoryWeather
+//
+//  Created by Matej Korman on 09/01/2019.
+//  Copyright © 2019 Matej Korman. All rights reserved.
+//
+
+import SnapKit
+
+protocol SettingsDisplayLogic: class {
+    func displayLocations(_ locations: [Location])
+    func initializeDisplay(locations: [Location], settings: Settings)
+}
+
+class SettingsViewController: UIViewController {
+    var interactor: SettingsBusinessLogic?
+    var router: SettingsRoutingLogic?
+    
+    private let contentView = SettingsContentView.autolayoutView()
+    private let activityIndicatorView = UIActivityIndicatorView.autolayoutView()
+    private var dataSource = SettingsDataSource()
+    
+    init(delegate: SettingsRouterDelegate?) {
+        super.init(nibName: nil, bundle: nil)
+        let interactor = SettingsInteractor()
+        let presenter = SettingsPresenter()
+        let router = SettingsRouter()
+        interactor.presenter = presenter
+        presenter.viewController = self
+        router.viewController = self
+        router.delegate = delegate
+        self.interactor = interactor
+        self.router = router
+        setupViews()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        interactor?.getInitailData()
+    }
+}
+
+// MARK: - Display Logic
+extension SettingsViewController: SettingsDisplayLogic {
+    func displayLocations(_ locations: [Location]) {
+        dataSource.setLocations(locations)
+        if let index = dataSource.sections.firstIndex(where: { $0 == SettingsSection.locations(rows: []) }) {
```

Comment:
Am I right to asume that you're just checking whether the `locations` section exists? If so, I think it would be more understandable if you exposed a method on the `data source` itself, that allows you to check whether the `locations` section is there.

P.s. What happens if we don't find an `index` here?

Reply:
I'm checking to see if the `locations` section exists, but more importantly I'm getting `sectionIndex` for reload. (We should always find an index because we're setting locations in the row above and even if there are no locations, we will get that section, because `dataSource` creates that section regardless. If we don't then there isn't any `locations` section, and there is nothing to be reloaded.)

Reply:
I see. I would still make a method on the `data source`, that returns this `index` - should be nicer as the `view controller` does not need to dig around the data source's section, and we have a well defined API on the data source.